### PR TITLE
[Feature] - Create useMousePosition hook

### PR DESCRIPTION
**Pull request by Lucas-Tisera**
## Description
Added `useMousePosition` hook to track mouse coordinates in real-time. This hook provides a simple way to access the current mouse position (x, y) coordinates anywhere in the application.

### Related Issue
Implements the mouse tracking hook requested in [#88](https://github.com/novajslabs/nova.js/issues/88)

### Features
- Real-time mouse position tracking
- Automatic cleanup of event listeners
- Zero dependencies
- Lightweight implementation

### Usage Example

```javascript
const { x , y } = useMousePosition();

console.log(Mouse is at: ${ x }, ${ y });
```

### Why
- Simplifies mouse position tracking across components
- Useful for interactive UI elements, animations, and custom cursors
- Follows React best practices with proper cleanup

### Technical Details
- Uses `window.addEventListener` for mouse tracking
- Implements proper cleanup in useEffect
- Returns coordinates as `{ x, y }` object

### CODE REVIEW

**Review by dlcastillop — CHANGES_REQUESTED** [review]
Great job, Lucas! Just check out the comments I made in the code. Thank you.

File: src/hooks/js/useMousePosition.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { useState, useEffect } from 'react';
+
+function useMousePosition() {
```
[the remark this answers is not in the document]
Comment (Lucas-Tisera) [reply src/hooks/js/useMousePosition.js:3]: mb, forgotten

File: src/hooks/js/useMousePosition.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  const [mousePosition, setMousePosition] = useState({
+    x: 0,
+    y: 0
+  });
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (dlcastillop) [reply src/hooks/js/useMousePosition.js:9]: Look at this implementation. It works for me.

```ts
import { useSyncExternalStore } from 'react';

interface MousePosition {
  x: number;
  y: number;
}

let mousePosition: MousePosition = { x: 0, y: 0 };

export const useMousePosition = (): MousePosition => {
  const subscribe = (callback: () => void) => {
    const handleMouseMove = (event: MouseEvent) => {
      mousePosition = {
        x: event.clientX,
        y: event.clientY
      };
      callback();
    };

    window.addEventListener('mousemove', handleMouseMove);
    return () => {
      window.removeEventListener('mousemove', handleMouseMove);
    };
  };

  const getSnapshot = (): MousePosition => mousePosition;

  const getServerSnapshot = (): MousePosition => ({ x: 0, y: 0 });

  return useSyncExternalStore(subscribe, getSnapshot, getServerSnapshot);
};
```

File: src/App.tsx
Comment (dlcastillop) [inline src/App.tsx:1]: Don´t modify this file.

**Review by dlcastillop — CHANGES_REQUESTED** [review]

File: src/hooks/js/useMousePosition.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lucas-Tisera) [reply src/hooks/js/useMousePosition.js]: It works 😁, thanks for the suggestion!

**Review by dlcastillop — APPROVED** [review]
Amazing job Lucas. Thank you!

**Pull request merged**